### PR TITLE
upgrade to selenium webdriver 3.4.0

### PR DESCRIPTION
**Pull request by ambirag**

### CODE REVIEW

File: src/main/java/com/rmn/qa/aws/AwsVmManager.java [hunk 1]
```diff
@@ -493,8 +493,11 @@ String getNodeConfig(final String uuid, final String hostName, final String brow
         Date createdDate = Calendar.getInstance().getTime();
 
         // Pass in the created date so we can know when this node was spun up
-        nodeConfig = nodeConfig.replaceAll("<CREATED_DATE>", AwsVmManager.NODE_DATE_FORMAT.format(createdDate));
-        nodeConfig = nodeConfig.replaceFirst("<HOST_NAME>", hostName);
+        String nodeCreatedDate = AwsVmManager.NODE_DATE_FORMAT.format(createdDate);
```
Comment (mhardin) [inline src/main/java/com/rmn/qa/aws/AwsVmManager.java]: So I have this code working locally with Selenium 3.0, and I'm not sure why you're having to put quotes in here?  Here is what I have locally that works:
```
        nodeConfig = nodeConfig.replaceAll("<CREATED_DATE>", AwsVmManager.NODE_DATE_FORMAT.format(createdDate));
        nodeConfig = nodeConfig.replaceFirst("<HOST_NAME>", hostName);
```

File: src/main/resources/node.linux.json [hunk 2]
```diff
@@ -4,27 +4,18 @@
...
-        "instanceId": "<INSTANCE_ID>",
-        "createdDate": "<CREATED_DATE>",
-        "createdBrowser": "<CREATED_BROWSER>",
-        "createdOs": "<CREATED_OS>",
         "register": true,
         "registerCycle": 5000,
-        "hubHost": "<HOST_NAME>"
-    }
+        "hub": "<HOST_NAME>",
+        "custom": {uuid=<UUID>,instanceId=<INSTANCE_ID>,createdDate=<CREATED_DATE>,createdBrowser=<CREATED_BROWSER>,createdOs=<CREATED_OS>}
```
Comment (mhardin) [inline src/main/resources/node.linux.json]: For the 'custom' field, can you have each field formatted to be on a new line?  Easier to read that way.

File: src/main/resources/node.linux.json [hunk 3]
```diff
@@ -4,27 +4,18 @@
             {
                 "browserName": "firefox",
                 "maxInstances": <MAX_SESSION_FIREFOX>,
-                "instanceId": "<INSTANCE_ID>",
                 "seleniumProtocol": "WebDriver"
             },
             {
                 "browserName": "chrome",
                 "maxInstances": <MAX_SESSION_CHROME>,
-                "instanceId": "<INSTANCE_ID>",
```
Comment (mhardin) [inline src/main/resources/node.linux.json]: You can leave these fields on the individual browser slots actually (have this working locally).

File: src/main/resources/node.windows.json [hunk 4]
```diff
@@ -4,33 +4,23 @@
             {
                 "browserName": "internet explorer",
                 "maxInstances": <MAX_SESSION_IE>,
-                "instanceId": "<INSTANCE_ID>",
```
Comment (mhardin) [inline src/main/resources/node.windows.json]: Same comment above as leaving these fields

File: src/main/resources/node.windows.json [hunk 5]
```diff
@@ -4,33 +4,23 @@
...
-        "instanceId": "<INSTANCE_ID>",
-        "createdDate": "<CREATED_DATE>",
-        "createdBrowser": "<CREATED_BROWSER>",
-        "createdOs": "<CREATED_OS>",
         "register": true,
         "registerCycle": 5000,
-        "hubHost": "<HOST_NAME>"
-    }
+        "hub": "<HOST_NAME>",
+        "custom": {uuid=<UUID>,instanceId=<INSTANCE_ID>,createdDate=<CREATED_DATE>,createdBrowser=<CREATED_BROWSER>,createdOs=<CREATED_OS>}
```
Comment (mhardin) [inline src/main/resources/node.windows.json]: Same comment as above about putting fields on new lines.

**Comment by mhardin** [comment]
@ambirag thanks for doing this.  Please address my comments and then I can merge this over.

File: src/main/java/com/rmn/qa/aws/AwsVmManager.java (hunk 1 above)
Comment (ambirag) [inline src/main/java/com/rmn/qa/aws/AwsVmManager.java]: when I tested this, the the node was created, the createdDate in nodeConfig.json had date like createdDate=05/25/2017 09:46:05 , but the node wouldn't start with json parsing error as selenium webdriver 3.4 I believe due to selenium using Jcommander. Once I put the dates between "" this started working fine.

Another note, current format of command  in node bootstrap code wouldn't start the node. All -D parameters should come before -jar option 
see here https://github.com/SeleniumHQ/selenium/issues/2566

Comment (mhardin) [inline src/main/java/com/rmn/qa/aws/AwsVmManager.java]: Hmmm.  So the date is in quotes as seen in the node.linux.json file, and I can confirm this is working locally with Selenium 3.0.0.  Can you double check your issue to see if A.) Its still reproducing 3.4.0, and if so, B.) If its just something with Selenium 3.4.0 (try with 3.0.0 and see if it reproduces)

**Comment by ambirag** [comment]
@mhardin

**Comment by ambirag** [comment]
cool, will make the change!

**Pull request merged**